helm/cluster-api/values.yaml

```diff
@@ -8,7 +8,7 @@ images:
     name: giantswarm/kubeadm-bootstrap-controller
   controlplane:
     name: giantswarm/kubeadm-control-plane-controller
-  tag: v1.4.0
+  tag: v1.4.1-gs-0f1c755e2
```

Comment:
This version is built from `main` branch in our fork, which means that this change makes `cluster-api-app` main branch unstable ([CAPI `main` branch is the development branch, not a stable branch](https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#branches)), and it can easily get deployed to production cluster because we make most of the releases from the `main` branch.

As described above, our apps and upstream CAPI projects use git branches in a different way, CAPI's `main` is the development branch (released once per quarter) and our CAPI app `main` branch is something where we use stable upstream CAPI and release it often.

While switching to using our fork is something we want (already discussed in KaaS Area in fork managements meets), using the main branch from our forks in the app `main` is dangerous ATM.
- It could be possible if we had some fork management automation, coupled with app release management automation and checks, so we're certain how we're releasing and deploying CAPI - and doing this in the same way across all CAPI apps (because provider apps depend on specific version of CAPI app).
- We would also need to adjust our git branch usage strategy to fully align with upstream, for which all new KaaS teams (Cluster Lifecycle, Cloud Integration, Onprem Integration) should talk about and agree, and again, do this consistently across all CAPI apps, which TBH can't imagine how that would work without a lot of missing automation and checks/validations that would ensure we don't do releases in ad-hoc ways.

Reply:
I'm changing this now so that `cluster-api` branch `release-1.4` (and also `main`) have instructions how to use stable releases. Using v1.4.2 + our CircleCI pipeline now, so it's a custom tag `v1.4.2-gs-...`.

You can review already, but I made _this PR_ a draft for now until it builds fine and I've re-tested it.